contracts/src/lunarswap-v1/LunarswapLibrary.compact

```diff
@@ -0,0 +1,428 @@
+    return [token0, token1];
+  }
+
+  /**
+   * @title addCoinValue circuit
+   * @description Adds a value to a coin and returns a new CoinInfo.
+   * 
+   * @remarks
+   * This circuit is functionally identical to addToCoin, creating a new coin with
+   * an increased value by adding the specified amount to the original coin's value.
+   *
+   * @circuitInfo k=11, rows=600
+   *
+   * @param {CoinInfo} _coin   - The original coin.
+   * @param {Uint<128>} _amount - The amount to add to the coin.
+   *
+   * @returns {CoinInfo} - A new coin with the increased value.
+   */
+  // Helper circuit to update a coin's value and return a new CoinInfo
+  export circuit addCoinValue(_coin: CoinInfo, _amount: Uint<128>): CoinInfo {
+    return CoinInfo { color: _coin.color,
+                      value: Uint128_addChecked(_coin.value, _amount),
+                      nonce: evolveNonce(0, _coin.nonce) };
+  }
+
+  /**
+   * @title addQualifiedCoinValue circuit
+   * @description Adds a value to a qualified coin and returns a new QualifiedCoinInfo.
+   * 
+   * @remarks
+   * This circuit creates a new qualified coin with an increased value by adding the
+   * specified amount to the original coin's value. The new coin maintains the same
+   * color and Merkle tree index but has a different nonce.
+   *
+   * @circuitInfo k=11, rows=700
+   *
+   * @param {QualifiedCoinInfo} _coin   - The original qualified coin.
+   * @param {Uint<128>} _amount - The amount to add to the coin.
+   *
+   * @returns {QualifiedCoinInfo} - A new qualified coin with the increased value.
+   */
+  export circuit addQualifiedCoinValue(coin: QualifiedCoinInfo, amount: Uint<128>): QualifiedCoinInfo {
+    return QualifiedCoinInfo { color: coin.color,
+                               value: Uint128_addChecked(coin.value, amount),
+                               nonce: evolveNonce(0, coin.nonce),
+                               mt_index: coin.mt_index };
+  }
+
+  /**
+   * @title subQualifiedCoinValue circuit
+   * @description Subtracts a value from a qualified coin and returns a new QualifiedCoinInfo.
+   * 
+   * @remarks
+   * This circuit creates a new qualified coin with a decreased value by subtracting the
+   * specified amount from the original coin's value. The new coin maintains the same
+   * color and Merkle tree index but has a different nonce.
+   *
+   * @circuitInfo k=11, rows=700
+   *
+   * @param {QualifiedCoinInfo} _coin   - The original qualified coin.
+   * @param {Uint<128>} _amount - The amount to subtract from the coin.
+   *
+   * @returns {QualifiedCoinInfo} - A new qualified coin with the decreased value.
+   */
+  export circuit subQualifiedCoinValue(coin: QualifiedCoinInfo, amount: Uint<128>): QualifiedCoinInfo {
+    assert(amount <= coin.value, "LunarswapLibrary: subQualifiedCoinValue() - Insufficient amount");
+    return QualifiedCoinInfo { color: coin.color,
+                               value: Uint128_sub(coin.value, amount),
```

Comment:
This is more of a math lib question, we have `Uint128_addChecked `, should this be `Uint128_subChecked `? Or is it supposed to be implied? I think in math, it should be the reverse: make safety the "default", and make danger loud e.g. `addUnchecked` and `add` (checked)